### PR TITLE
SONARAZDO-408 Use strict: true in our tsconfig file

### PR DESCRIPTION
**Pull request by 7PH**
This was spotted during the implementation of a ticket in this sprint - Hopefully, this allows us to spot errors early in the future!

### CODE REVIEW

File: src/common/sonarcloud-v1/helpers/azdo-api-utils.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function parseScannerExtraProperties(): { [key: string]: string } {
   const props = {};
   tl.getDelimitedInput("extraProperties", "\n")
     .filter((keyValue) => !keyValue.startsWith("#"))
-    .map((keyValue) => keyValue.split(/=(.+)/))
+    .map((keyValue) => keyValue.split(/[=](.+)/))
```
Comment (7PH) [inline src/common/sonarcloud-v1/helpers/azdo-api-utils.ts:68]: I don't know why but eslint started failing on this now

File: .eslintrc [hunk 2]
```diff
@@ -39,7 +39,6 @@
     "guard-for-in": "error",
     "no-alert": "error",
     "no-caller": "error",
-    "no-div-regex": "warn",
```
Comment (7PH) [inline .eslintrc:42]: This conflicts with a SonarQube rule

File: src/common/latest/helpers/__tests__/api-test.ts [hunk 3]
```diff
@@ -73,7 +73,7 @@ describe("fetchWithRetry", () => {
       expect.assertions(2);
       try {
         await fetchWithRetry(MOCKED_ENDPOINT, "/api");
-      } catch (error) {
+      } catch (error: any) {
```
Comment (lucas-paulger-sonarsource) [inline src/common/latest/helpers/__tests__/api-test.ts:76]: typically unknown is preffered over any
```suggestion
      } catch (error: unknown) {
```

**Review by lucas-paulger-sonarsource — APPROVED** [review]
LGTM! nice work, should save us significant time

File: src/common/latest/helpers/__tests__/api-test.ts (hunk 3 above)
Comment (7PH) [reply src/common/latest/helpers/__tests__/api-test.ts]: doesn't hurt to be consistent using `error: unknown`, so i've changed it 👍

**Pull request merged**